pyOCD/coresight/ap.py

```diff
@@ -100,7 +100,21 @@ def write_reg(self, addr, data):
 class MEM_AP(AccessPort):
     def __init__(self, dp, ap_num):
         super(MEM_AP, self).__init__(dp, ap_num)
-        self.auto_increment_page_size = 4
+
+        # Default to the smallest size supported by all targets.
+        # A size smaller than the supported size will decrease performance
+        # due to the extra address writes, but will not create any
+        # read/write errors.
+        self.auto_increment_page_size = 0x400
+
+    def init(self, bus_accessible=True):
+        super(MEM_AP, self).init(bus_accessible)
+
+        # Look up the page size based on AP ID.
+        try:
+            self.auto_increment_page_size = MEM_AP_IDR_TO_WRAP_SIZE[self.idr]
```

Comment:
Why is a try-except block used here? In the previous version there was a check if self.idr was in MEM_AP_IDR_TO_WRAP_SIZE.